### DISCUSSION TITLE
A new config file for storing credentials

### DISCUSSION THREAD

**Discussion by lfv89** [ruby/rubygems#6136] [Ideas]
### Problem:

At the moment the `~/.bundle/config` stores both credentials and global configuration as well.

This is not ideal for versioning purposes. It does makes sense to version the file because global configurations like `BUNDLE_PATH` are useful across multiple setups, but it's not safe to do it right now because credentials for accessing private gems (like `BUNDLE_GEMS__CONTRIBSYS__COM` for Sidekiq) would also be versioned, which is not desirable.

### Proposal:

These credentials should go to their own `~/.bundle/credentials` file which would greatly help to avoid them getting leaked from public repositories, since there would be no point to version this specifically file.

---

I'd gladly jump into this if the core team finds it worthy.

Let me know what you think.

**Comment by simi** [comment]
Seems like great candidate for [RFC](https://github.com/rubygems/rfcs). Crucial would be to do that only for new installations and support the old way of storing credentials when still present.

**Comment by deivid-rodriguez** [comment]
Moved to a discussion until it becomes something more actionable or gets a RFC written for it.
